storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2698,7 +2400,9 @@ static bool innodb_init(bool init_dd, bool for_apply_log) {
     if (dict_persist->table_buffer == nullptr)
       dict_persist->table_buffer =
           ut::new_withkey<DDTableBuffer>(UT_NEW_THIS_FILE_PSI_KEY);
-    srv_dict_recover_on_restart();
+    if (!srv_apply_log_only) {
```

Comment:
Yes, it is a mistake before to do rollback of dictionary trxs in apply_log only mode.

Regarding the for_apply_log, it seems the callers are

```
Functions calling this function: innodb_init

  File          Function                Line
0 xtrabackup.cc xtrabackup_stats_func   4735 if (innodb_init(true, false)) exit(EXIT_FAILURE);
1 xtrabackup.cc xtrabackup_prepare_func 6991 if (innodb_init(true, true)) {
```
so they are not same. May we didn't want to initiate recovery when operating on server log data directory with redo files..
Name could have been better

Reply:
we will need an extra patch to catch the redo log generated during apply log phase.  Since it is touching the dictionary-related area, I fixed this too.